source/extensions/tracers/xray/xray_tracer_impl.cc

```diff
@@ -1,44 +1,108 @@
 #include "extensions/tracers/xray/xray_tracer_impl.h"
 
 #include "common/common/macros.h"
+#include "common/common/utility.h"
+
+#include "extensions/tracers/xray/localized_sampling.h"
+#include "extensions/tracers/xray/tracer.h"
+#include "extensions/tracers/xray/xray_configuration.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace Tracers {
 namespace XRay {
 
-static const char DefaultDaemonEndpoint[] = "127.0.0.1:2000";
-Driver::Driver(const XRayConfiguration& config, Server::Instance& server) : xray_config_(config) {
+constexpr static char DefaultDaemonEndpoint[] = "127.0.0.1:2000";
+static XRayHeader parseXRayHeader(const Http::LowerCaseString& header) {
+  const auto& lowered_header = header.get();
+  XRayHeader result;
+  for (const auto& token : StringUtil::splitToken(lowered_header, ";")) {
+    if (token.find("root=") == 0) {
```

Comment:
Since `token` is an `absl::string_view` we could use `absl::StartsWith(token, "root=")` here instead of general string search.